tls/s2n_client_hello.c

```diff
@@ -423,6 +423,8 @@ int s2n_process_client_hello(struct s2n_connection *conn)
         conn->actual_protocol_version = MIN(conn->server_protocol_version, conn->client_protocol_version);
     }
 
+    POSIX_GUARD_RESULT(s2n_conn_choose_state_machine(conn, conn->actual_protocol_version));
```

Comment:
`s2n_process_client_hello` is only called if client_hello_version != S2N_SSLv2:
https://github.com/aws/s2n-tls/blob/4a739197de0d7121ac84189c79854c64ad37da6e/tls/s2n_client_hello.c#L535-L537

Should this call to `s2n_conn_choose_state_machine` be moved to somewhere where it will always be called?

Reply:
Technically, since the state machine macro treats STATE_MACHINE_INITIAL as equivalent to STATE_MACHINE_TLS12, it doesn't matter if we don't actually set the state machine here.

Reply:
Right okay, that makes sense. I guess it just seems weird to have it be set to initial for the whole handshake potentially, since it might be able to swap to 1.3 at some point. But I guess since this is the only place it's set that's not an issue?

Reply:
Nevermind, I should probably fix this, see @lrstewart's comment below